build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/types/DockerImage.java

```diff
@@ -207,64 +206,83 @@ public Module generateBuildInfoModule(ArtifactoryBuildInfoClientBuilder buildInf
                 return buildInfoModule;
             }
             logger.info("Fetching details of published docker layers from Artifactory...");
-            setDependenciesAndArtifacts(buildInfoModule, buildInfoItemsProps, artifactsPropsStr,
-                    dependenciesClient, buildInfoClient);
+            if (cmdType == DockerUtils.CommandType.Push) {
+                setDependenciesAndArtifacts(buildInfoModule, dependenciesClient);
+            } else {
+                setDependencies(buildInfoModule);
+            }
             setBuildInfoModuleProps(buildInfoModule);
             return buildInfoModule;
         }
     }
 
+    private void loadLayers() throws IOException {
+        try (ArtifactoryBuildInfoClient buildInfoClient = buildInfoClientBuilder.build();
+             ArtifactoryDependenciesClient dependenciesClient = dependenciesClientBuilder.build()) {
+            this.layers = this.getLayers(dependenciesClient, buildInfoClient);
+            List<DockerLayer> markerLayers = this.layers.getLayers().stream().filter(layer -> layer.getFileName().endsWith(".marker")).collect(Collectors.toList());
+            if (markerLayers.size() > 0) {
+                for (DockerLayer markerLayer : markerLayers) {
+                    dependenciesClient.downloadMarkerLayer(targetRepo, this.imageTag.substring(imageTag.indexOf("/") + 1, imageTag.indexOf(":")), markerLayer.getDigest());
+                }
+                this.layers = this.getLayers(dependenciesClient, buildInfoClient);
```

Comment:
`this` is redundant. Please use only in getters/setters/when there is no other choice.
```suggestion
                layers = getLayers(dependenciesClient, buildInfoClient);
```